### PR TITLE
Add state and related to entity reg dialog

### PR DESCRIPTION
**Pull request by bramkragten**
Update to match `more-info` and replace `more-info-settings`
image

image

### CODE REVIEW

**Comment by iantrich** [comment]
Very nice!

File: src/components/ha-related-items.ts [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+            if (!entry) {
+              return;
+            }
+            return html`
+              <h3>
+                ${this.hass.localize(
+                  "ui.components.related-items.integration"
+                )}:
+              </h3>
+              <a
```
Comment (balloob) [inline src/components/ha-related-items.ts]: So not, now, but for a future PR:

We should create web components that render a config entry, device, entity, area etc as linkable entities. Then we can also start adding hover tooltips ala GitHub (hover over my name) in the future.

File: src/data/search.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  | "area"
+  | "automation"
+  | "config_entry"
+  | "device"
+  | "entity"
+  | "group"
+  | "scene"
+  | "script";
+
+export const findRelated = (
```
Comment (balloob) [inline src/data/search.ts]: Yeeesssssssssssssssss 👌

File: src/panels/config/entities/ha-config-entities.ts [hunk 3]
```diff
@@ -196,10 +196,23 @@ export class HaConfigEntities extends LitElement {
...
+        this._entities = entities;
+      }),
+    ];
+  }
+
   public disconnectedCallback() {
     super.disconnectedCallback();
-    if (this._unsubEntities) {
-      this._unsubEntities();
+    const dialog = document
```
Comment (balloob) [inline src/panels/config/entities/ha-config-entities.ts]: Eeeh, this is not nice. Can't we return a function from `showXDialog` that allows closing it ?

**Comment by iantrich** [comment]
Beautiful addition! Very nice @bramkragten 🎉

**Comment by balloob** [comment]
I think that we should remove the `state` tab from the entity registry dialog and have a button on the top right to switch back to the more info. That would probably feel more natural.

**Comment by bramkragten** [comment]
> I think that we should remove the `state` tab from the entity registry dialog and have a button on the top right to switch back to the more info. That would probably feel more natural.

What PR did you review? 😛 that's what it does now (the top screenshot).

File: src/panels/config/entities/ha-config-entities.ts (hunk 3 above)
Comment (bramkragten) [reply src/panels/config/entities/ha-config-entities.ts]: Yeah this is ugly... that sounds good will try 👍

**Comment by balloob** [comment]
what's the 2nd screenshot then?

File: src/panels/config/entities/ha-config-entities.ts (hunk 3 above)
Comment (balloob) [reply src/panels/config/entities/ha-config-entities.ts]: You can put this exact logic in `showXDialog`, but that should be the only place that knows about the HTML tag name of the dialog element.

**Review by balloob — APPROVED** [review]

**Comment by balloob** [comment]
If the 2nd screenshot is the entity registry dialog, then we should merge that with the more info settings dialog (if that makes sense). 

- So from more info, you click settings -> opens entity settings dialog.
- From entity settings dialog you click the state icon in top right -> more info dialog

**Comment by bramkragten** [comment]
The second screenshot is old, changed it today to what you just said. So no states tab in entity reg dialog and no settings in more info dialog, they both have a button to open the other dialog.

**Pull request merged**